logstash-core/lib/logstash/instrument/metric.rb

```diff
@@ -45,20 +57,14 @@ def time(key, &block)
       end
     end
 
-    # TODO
-    def self.create(name, collector = LogStash::Instrument::Collector.instance)
+    def self.create(namespace, collector = LogStash::Instrument::Collector.instance)
       reporter = LogStash::Instrument::Reporter::Stdout.new(collector)
-      Metric.new(collector, name)
+      Metric.new(collector, namespace)
     end
 
     private
-    def merge_keys(key)
-      valid_key!(key)
-      [@base_key, key.to_sym]
-    end
-    
-    def valid_key!(key)
-      raise MetricNoKeyProvided if key.nil? || key == ""
+    def validate_key!(key)
+      raise MetricNoKeyProvided if key.nil? || key.empty?
```

Comment:
If I understand correctly you are talking about type checking for the collected metric?
If that the case yes this is a problem with the actual code! +1

Lets say that we have a key of "root.pipeline01.events_in" of type `Counter` and we try to use another type with the same key it should raise an error? Correct?

One way to fix that issue could be to add the type to the actual key automatically, so the previous key would become.

`root.pipeline01.events_in.counter` ?